apis/v1alpha2/shared_types.go

```diff
@@ -263,6 +263,58 @@ type RouteStatus struct {
 	Parents []RouteParentStatus `json:"parents"`
 }
 
+// RouteParentStatusWithHostnames describes the status of a route with respect
+// to an associated Parent. In the experimental channel, this includes hostnames
+// used as part of the Parent binding.
+type RouteParentStatusWithHostnames struct {
```

Comment:
The `WithHostnames` suffix feels a little verbose, but I can't think of anything better that's not worse. 😞

Reply:
Haha same.